static/js/hash_util.js

```diff
@@ -310,3 +310,32 @@ export function build_login_link() {
     }
     return login_link;
 }
+
+export function get_current_view_type() {
+    const hash = window.location.hash.split("/");
+    try {
+        const operators = parse_narrow(hash);
+        if (
+            operators.length === 1 &&
+            !operators[0].negated &&
+            operators[0].operator === "pm-with"
+        ) {
+            return "pm";
+        }
+        if (operators.length === 1 && !operators[0].negated && operators[0].operator === "stream") {
+            return "stream";
+        }
+        if (
+            operators.length === 2 &&
+            !operators[0].negated &&
+            operators[0].operator === "stream" &&
+            !operators[0].negated &&
+            operators[1].operator === "topic"
+        ) {
+            return "topic";
+        }
+        return "other";
+    } catch {
+        return "other";
+    }
+}
```

Comment:
This should be replaced with calling some functions accessing the `Filter` object that is `narrow_state.filter` -- we try to avoid parsing URLs in additional places, since it inevitably leads to bugs. I've not checked if existing methods on that can do this exactly or if you need to add something.

Reply:
Thanks for the hint! I found suitable functions using `Filter` in `narrow_state.js` so have used those directly in `hotkey.js`